core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -246,35 +237,52 @@ public TableToken createTable(
         return tableToken;
     }
 
-    // caller has to acquire the lock before this method is called and release the lock after the call
-    public void createTableUnsafe(
+    public TableToken createTable(
             CairoSecurityContext securityContext,
             MemoryMARW mem,
             Path path,
+            boolean ifNotExists,
             TableStructure struct,
-            TableToken tableToken,
-            int tableId
+            boolean keepLock
     ) {
-        securityContext.checkWritePermission();
+        return createTableExecutor(
+                securityContext,
+                mem,
+                path,
+                ifNotExists,
+                struct,
+                keepLock,
+                tableToken -> {
```

Comment:
Lambdas create instances e.g. garbage. We do not use lamas for same of the code better look.

Reply:
addressed in b79b44074e4e53a96eb9c4d3e548f901fc1f2188